src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -493,6 +493,9 @@ private void RunTestCleanupMethod(object classInstance, TestResult result)
             }
             finally
             {
+#if NET6_0_OR_GREATER
```

Comment:
It would be nice to capture the reasoning we had about why we think it's a good idea to keep the call to `Dispose()` after the call to `DisposeAsync()`. This will help us and others in the future :)

As a "general" rule, if you are doing something that is not straightforward, it's always better to explain the reasoning. It helps when doing review and when debugging/investigating later.